src/headline_grabber/__main__.py

```diff
@@ -42,25 +42,92 @@
 @click.option(
     "--limit",
     "-l",
-    type=str,
+    type=int,
     default=None,
     required=False,
     callback=OptionValidator.validate_max_entries,
     help="Number specifying the maximum number of entries per topic in a report",
 )
-def main(include: str, exclude: str, target_dir: str, limit: int):
+@click.option(
+    "--interactive",
+    is_flag=True,
+    default=False,
+    help="Launch interactive menu for preference selection",
+)
+def main(include: str, exclude: str, target_dir: str, limit: int, interactive: bool):
     """Simple program to collect headlines from various news sources and summarize them in a helpful way"""
+    if interactive:
+        user_preferences = run_interactive_menu()
+    else:
+        user_preferences = UserPreferences(
+            include=(include.split(",") if include else None),
+            exclude=(exclude.split(",") if exclude else None),
+            target_dir=target_dir if target_dir else None,
+            limit=limit if limit else None,
+        )
     pipeline_context = PipelineContext(
         site_configs=sites,
         headlines=[],
         grouped_headlines={},
         documents_for_display={},
-        user_input=UserPreferences(
-            include=(include.split(",") if include else None),
-            exclude=(exclude.split(",") if exclude else None),
-            target_dir=(target_dir if target_dir else None),
-            limit=(limit if limit else None),
-        ),
+        user_input=user_preferences,
     )
     pipeline_context = news_pipeline.run(pipeline_context)
 
+def run_interactive_menu() -> UserPreferences:
```

Comment:
While implementing simple term menu, i was getting errors. After researching more about the error, i found that simple_term_menu is not supported on Windows because it relies on Unix-specific modules like termios, which are not available on Windows systems.